### ISSUE TITLE
2D images "swallow" points on the same plane if viewed in 3D

### ISSUE THREAD

**Issue by brisvag** [napari/napari#4402]
## 🐛 Bug

With vispy 0.10, `VolumeVisual` is correctly setting fragment depth (instead of leaving it unset). This, combined with the fact that we use a `Volume` of thickness `1 px` to display `Image`s in 3D, caused #4399. This is because a point is technically *inside* the 1 pixel of the Volume, rather than "on" it, like we would expect.

This is a problem because People might want to look at 2D data in 3D with the associated points, but suddenly the points appear to disappear (:facepalm:).

Try out this code:

```python
import napari
import numpy as np

v = napari.Viewer()
v.add_image(np.ones((100, 100)))
pl = v.add_points([50, 50], face_color='r')
```

Switch to 3D and the point will disappear. If you rotate the view, however, it will be clear that the rendering is behaving as intended for a thin slab of volume.

Workarounds for the time being:

1. use the spherical shading mode, since this also sets the depth buffer:
```
pl.shading = 'spherical'
```
This is not necessarily good for every application, it looks bad with the `edge` active, and it's size-dependent (a small point will still disappear).

2. use the `translucent_no_depth` blending on one of the layers.

Possible solution: use actual `ImageVisual` *even in 3D* when the data is 2D. This requires some reworking and refactoring of the `VispyImageLayer` to account for this new case, but to me it seems the only reasonable way to solve this issue.

On the other hand, it might just be "right" to accept this new behaviour as "correct" :P

## Additional context

- #4401

**Comment by jni** [comment]
Thanks @brisvag for investigating and writing this up! 

> Possible solution: use actual ImageVisual even in 3D when the data is 2D. This requires some reworking and refactoring of the VispyImageLayer to account for this new case, but to me it seems the only reasonable way to solve this issue.

I agree here. The issue is that the image in this case really should have 0 thickness since it is 2D.

**Comment by psobolewskiPhD** [comment]
Yeah, if there are only 2 axes then there shouldn't be a thickness—should the 3D viewer even work in that case? Maybe just grey-out the button with a hover message?
If there is a singleton 3rd axis, say z, then I can see rendering the voxels, plus in theory there could be a scale associated with that 3rd axis...

**Comment by jni** [comment]
> Yeah, if there are only 2 axes then there shouldn't be a thickness—should the 3D viewer even work in that case? Maybe just grey-out the button with a hover message?

We still need to display a 2D image in 3D in the case of mixed-dimensionality data.

**Comment by psobolewskiPhD** [comment]
I tested the script from the OP and the point remains visible in 3D:
<img width="1198" alt="image" src="https://github.com/user-attachments/assets/49b03400-563d-4cb1-a622-b782c90f4da7">
@brisvag was this fixed on vispy side maybe?

**Comment by brisvag** [comment]
It is indeed because of `translucent_no_depth`. If you use `translucent` you get this:

![image](https://github.com/user-attachments/assets/9007836c-daaa-4bda-bc18-f976ecd66d5c)

So it's the same as OP. This cannot be fixed on the vispy side, it's "correct" that the point gets hidden in 3D. What we can do is to change from Volume to Image node even in 3D when we have 2D data, which would clip only half the point, not more than half. Or use `translucent_no_depth`.
